compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -533,34 +516,25 @@ generic32BitAddEvaluator(TR::Node * node, TR::CodeGenerator * cg)
       int32_t value;
       switch (secondChild->getDataType())
          {
-         case TR::Address:
-            TR_ASSERT( TR::Compiler->target.is32Bit(),"Should not be here for 64bit!");
-         case TR::Int32:
+         case TR::Int8:
             {
-            PRINT_ME("iconst", node, cg);
-            value = secondChild->getInt();
+            value = (int32_t) secondChild->getByte();
             break;
             }
          case TR::Int16:
             {
-            PRINT_ME("sconst", node, cg);
             value = (int32_t) secondChild->getShortInt();
             break;
             }
-//          case TR_UInt16:
-//             {
-//             PRINT_ME("cconst", node, cg);
-//             value = (int32_t) secondChild->getConst<uint16_t>();
-//             break;
-//             }
-         case TR::Int8:
+         case TR::Int32:
             {
-            PRINT_ME("bconst", node, cg);
-            value = (int32_t) secondChild->getByte();
+            value = secondChild->getInt();
             break;
             }
+         case TR::Address:
+            TR_ASSERT_FATAL(TR::Compiler->target.is32Bit(), "Unexpected address data type for generic32BitAddEvaluator");
```

Comment:
In the old implementation, the case for `TR::Address` falls through to the `TR::Int32` case (assuming the assert passes) because of the missing `break;`. However, with this change, the `TR::Address` cases falls through to the `default` case that will unconditionally fatal assert. Is this change in behaviour intentional?

Reply:
I did check this in fact. The `generic32BitAddEvaluator` is called from the following functions:

- `iaddEvaluator`
- `baddEvaluator`
- `saddEvaluator`
- `caddEvaluator`
- `addrAddHelper`

Out of these the only one with a `TR::Address` should be coming from `addrAddHelper` which does not have any callers. This function is dead. I checked the tree evaluator table [1] and the address add evaluators have their own separate implementation.

There seems to be a lot of duplicate code here which should get cleaned up. I'll open up an issue for this. Incidentally though had there been a mistake it would have been caught the Tril test in [2] I believe. Do we have explicit address add/sub tests anywhere? I thought I saw a PR at some point?

[1] https://github.com/eclipse/omr/blob/4cee14984f633e4229ac2edbb12d19d0954d6847/compiler/z/codegen/TreeEvaluatorTable.hpp#L653-L654

[2] https://github.com/eclipse/omr/blob/f8fc704032f95cbc0f337eb6be54f2b06e58a773/fvtest/compilertriltest/ILValidatorTest.cpp#L77

Reply:
I'm going to actually add another commit to remove the `TR::Address` case and fall down to the default until the cleanup is performed to handle all of this correctly.

Reply:
> I thought I saw a PR at some point?

Ah of course it is my vague memory. It was not a PR but rather an issue opened by yours truly in #2922. All the more reason to hope on that. I'll see if I can find some time for it.

Reply:
Actually this switch case is for extracting constant from the second child of the node which should be integer for `aiadd`so even if this is called from `addrAddHelper`, second child should be int . We should not call this function when we are adding constant address.  I agree with @fjeremic , It should be unexpected type and fall down to default

Reply:
Fixed in 86b86e8 and opened #3668.